scripts/CMakeLists.txt

```diff
@@ -400,6 +400,89 @@ function(dmod_add_library moduleName version)
     dmod_create_module(${moduleName} ${version} Library ${SOURCES})
 endfunction()
 
+#
+# Function to link external modules and download their headers
+#
+# Usage:
+#   dmod_link_modules(target_name
+#       module1@1.0
+#       module2
+#       module3@2.1
+#   )
+#
+# This function downloads headers for the specified modules using dmf-get
+# and adds them to the target's include directories.
+#
+function(dmod_link_modules targetName)
+    set(MODULE_SPECS ${ARGN})
+    
+    # Check if target exists
+    if(NOT TARGET ${targetName})
+        message(FATAL_ERROR "dmod_link_modules: Target '${targetName}' does not exist")
+    endif()
+    
+    # Find dmf-get tool
+    find_program(DMF_GET 
+        dmf-get
+        HINTS ${DMOD_TOOLS_BIN_DIR}
+        PATHS ENV PATH
+    )
+    
+    if(NOT DMF_GET)
+        message(WARNING "dmf-get tool not found. Skipping module headers download for ${targetName}. Build the tools first with -DDMOD_BUILD_TOOLS=ON")
+        return()
+    endif()
+    
+    message(STATUS "Using dmf-get: ${DMF_GET}")
+    
+    # Get the default DMF directory for headers
+    if(NOT DEFINED DMOD_DMF_DIR)
+        set(DMOD_DMF_DIR "${CMAKE_BINARY_DIR}/dmf")
+    endif()
+    
+    # Process each module specification
+    foreach(MODULE_SPEC ${MODULE_SPECS})
+        # Parse module name and version
+        string(REGEX MATCH "^([^@]+)(@(.+))?$" MATCH_RESULT "${MODULE_SPEC}")
+        set(MODULE_NAME "${CMAKE_MATCH_1}")
+        set(MODULE_VERSION "${CMAKE_MATCH_3}")
+        
+        if(NOT MODULE_NAME)
+            message(WARNING "Invalid module specification: ${MODULE_SPEC}")
+            continue()
+        endif()
+        
+        # Construct the header output directory
+        set(HEADER_OUTPUT_DIR "${DMOD_DMF_DIR}/${MODULE_NAME}/inc")
+        
+        message(STATUS "Downloading headers for module: ${MODULE_NAME} -> ${HEADER_OUTPUT_DIR}")
+        
+        # Create the output directory
+        file(MAKE_DIRECTORY "${HEADER_OUTPUT_DIR}")
+        
+        # Run dmf-get headers command
+        execute_process(
+            COMMAND ${DMF_GET} headers ${MODULE_SPEC} -o ${HEADER_OUTPUT_DIR}
```

Comment:
Powinieneś też przekazać tools_name które zostało przekazane w parametrach do cmake, jeśli jest. Zwykle pewnie nie będzie miało to znaczenia, bo to headery, ale jestem w stanie wyobrazić sobie przypadek w którym to ma znaczenie